src/main/java/com/google/devtools/build/lib/actions/Spawns.java

```diff
@@ -33,6 +33,25 @@ public static boolean mayBeCached(Spawn spawn) {
         && !spawn.getExecutionInfo().containsKey(ExecutionRequirements.LOCAL);
   }
 
+  /**
+   * Returns {@code true} if the result of {@code spawn} may be cached remotely.
+   */
+  public static boolean mayBeCachedRemotely(Spawn spawn) {
```

Comment:
If the both `no-remote-cache` and `no-cache` are essentially the same for now (this PR is not about changing the `no-cache` behavior, right?) I'd suggest to make `mayBeCachedRemotely` and `mayBeCached` the same for now and add a comment to the latter, that it probably will change. wdyt?

Reply:
I was thinking that `no-remote-cache` wouldn't apply to the disk cache, since that's technically a local cache. We do refer to it as a remote cache in many places though, so maybe I should still treat that as a remote cache. If that's the case, then `no-remote-cache` and `no-cache` are currently identical.

Should I keep it as I have it, or should I treat the disk cache as a remote cache and remove `no-remote-cache`?

Reply:
Gotcha,
I'd say that disk cache is better to be treated as a remote cache, we call `local cache` something else. That would also make everything more consistent. @buchgr wdyt?

Reply:
While the disk cache is an implementation of the remote cache interfaces it's not a remote cache. I think intuitively users would expect no-remote and no-remote-cache to not apply to the disk cache. I believe there is also a bug about this somewhere.

Reply:
Ok, then I think we need to write it down for the future references:
no-cache - forbids remote (both http and grpc) and no local(disk) cache
no-remote-cache - forbids remote (both http and grpc) cache only
no-remote-exec - forbids remote execution

Looks good? :)

Reply:
perfect. thanks!

Reply:
Awesome. In this case, I'll leave it as I have it. I'll address the other comments @ishikhman left